subsys/bluetooth/host/scan.c

```diff
@@ -1768,13 +1810,10 @@ int bt_le_per_adv_sync_create(const struct bt_le_per_adv_sync_param *param,
 	 * established. We don't need to use any callbacks since we rely on
 	 * the advertiser address in the sync params.
 	 */
-	if (!atomic_test_bit(bt_dev.flags, BT_DEV_SCANNING)) {
-		err = bt_le_scan_update(true);
-
-		if (err) {
-			bt_le_per_adv_sync_delete(per_adv_sync);
-			return err;
-		}
+	err = bt_le_scan_update_and_reconfigure(SCAN_ENABLED_REASON_SYNC_SYNCING, true);
```

Comment:
The function name doesn't really give any hints what `true` or `false` mean for the last parameter. Boolean inputs are fine when the function name is boolean-style in itself, but here it's a bit fuzzy what the function might do, simply by looking at the function call itself.

I'd instead propose to use `uint8_t` and the existing `BT_HCI_LE_SCAN_DISABLE`/`BT_HCI_LE_SCAN_ENABLE` defines. The implementation of the function will still stay exactly the same (since you're just assigning this to an HCI command parameter) but the calls to the function will become clearer in their meaning.

Reply:
good point, will update